src/libraries/System.Text.Json/src/System/Text/Json/ThrowHelper.cs

```diff
@@ -723,6 +714,7 @@ internal enum NumericType
     internal enum DataType
     {
         Boolean,
+        DateOnly,
```

Comment:
Why don't we need TimeOnly here and in the switch above?

Reply:
`TimeOnly` delegates parsing to the `TimeSpan` converter. I considered using a separate message but I don't think it's important enough to warrant code duplication.